ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/TextEmbeddingModel.java

```diff
@@ -0,0 +1,239 @@
+                                       MLModelConfig modelConfig,
+                                       String engine) {
+        try {
+            if (FunctionName.TEXT_EMBEDDING != functionName) {
+                throw new IllegalArgumentException("wrong function name");
+            }
+            if (!PYTORCH_ENGINE.equals(engine) && !ONNX_ENGINE.equals(engine)) {
+                throw new IllegalArgumentException("unsupported engine");
+            }
+            AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+                ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+                try {
+                    System.setProperty("PYTORCH_PRECXX11", "true");
+                    System.setProperty("DJL_CACHE_DIR", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    System.setProperty("java.library.path", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    Thread.currentThread().setContextClassLoader(ai.djl.Model.class.getClassLoader());
+                    Path modelPath = getModelCachePath(modelId, modelName, version);
+                    File pathFile = new File(modelPath.toUri());
+                    if (pathFile.exists()) {
+                        FileUtils.deleteDirectory(pathFile);
+                    }
+                    ZipUtils.unzip(new FileInputStream(modelZipFile), modelPath);
+                    boolean findModelFile = false;
+                    for (File file : pathFile.listFiles()) {
+                        String name = file.getName();
+                        if (name.endsWith(PT) || name.endsWith(ONNX)) {
+                            if (findModelFile) {
+                                throw new IllegalArgumentException("found multiple models");
+                            }
+                            findModelFile = true;
+                            String suffix = name.substring(name.lastIndexOf("."));
+                            file.renameTo(new File(modelPath.resolve(modelName + suffix).toUri()));
+                        }
+                    }
+                    Map<String, Object> arguments = new HashMap<>();
+                    arguments.put("engine", engine);
+                    Criteria.Builder<Input, Output> criteriaBuilder = Criteria.builder()
+                            .setTypes(Input.class, Output.class)
+                            .optApplication(Application.UNDEFINED)
+                            .optArguments(arguments)
+                            .optModelPath(modelPath);
+                    TextEmbeddingModelConfig textEmbeddingModelConfig = (TextEmbeddingModelConfig) modelConfig;
+                    TextEmbeddingModelConfig.FrameworkType transformersType = textEmbeddingModelConfig.getFrameworkType();
+                    if (ONNX_ENGINE.equals(engine)) { //ONNX
```

Comment:
Why use two different type of model. They are actually the same, just with one more layer.

Reply:
Yes, agree, ideally they should be the same, we still met some issue for ONNX result. Seems its result can't match 
 Python model. This ONNX translator is for debugging that part. Once that testing done, we may merge it with other translator.

Reply:
It looks for MODEL_NAME.pt file in the model directory, if modelName is not specified in the criteria, it assume the model has the same name as the directory.

Reply:
Thanks, that logic seems not so user friendly as they have to make sure model name matches their `.pt` file name. We add rename logic to make things easier: `file.renameTo(new File(modelPath.resolve(modelName + suffix).toUri()));` . So user can use `.pt` file with any name, we will rename it internally to `<model_name>.pt` when load model. That means we have to make sure there is only one `.pt` file in the zip, otherwise we don't know which `.pt` file we should rename.